geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/list/AbstractLTrimIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+  private JedisCluster jedis;
+
+  @Before
+  public void setUp() {
+    jedis = new JedisCluster(new HostAndPort(BIND_ADDRESS, getPort()), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void givenWrongNumOfArgs_returnsError() {
+    assertExactNumberOfArgs(jedis, Protocol.Command.LTRIM, 3);
+  }
+
+  @Test
+  public void withNonListKey_Fails() {
+    jedis.set("string", PREEXISTING_VALUE);
+    assertThatThrownBy(() -> jedis.ltrim("string", 0, -1))
+        .hasMessage(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void withNonExistentKey_returnsOK() {
+    assertThat(jedis.ltrim("nonexistent", 0, -1)).isEqualTo("OK");
+  }
+
+  @Test
+  public void withNonIntegerRangeSpecifier_Fails() {
+    jedis.lpush(KEY, "e1", "e2", "e3", "e4");
+
+    assertThatThrownBy(() -> jedis.sendCommand(KEY, Protocol.Command.LTRIM, KEY,
+        "0", "not-an-integer"))
+            .hasMessage(ERROR_NOT_INTEGER);
+    assertThatThrownBy(() -> jedis.sendCommand(KEY, Protocol.Command.LTRIM, KEY,
+        "not-an-integer", "-1"))
+            .hasMessage(ERROR_NOT_INTEGER);
+    assertThatThrownBy(() -> jedis.sendCommand(KEY, Protocol.Command.LTRIM, KEY,
+        "not-an-integer", "not-an-integer"))
+            .hasMessage(ERROR_NOT_INTEGER);
+  }
+
+  @Test
+  @Parameters(method = "getValidRanges")
+  @TestCaseName("{method}: start:{0}, end:{1}, expected:{2}")
+  public void trimsToSpecifiedRange_givenValidRange(long start, long end, String[] expected) {
+    initializeTestList();
+
+    jedis.ltrim(KEY, start, end);
+    assertThat(jedis.lrange(KEY, 0, -1)).containsExactly(expected);
+  }
+
+  // @Parameterized.Parameters(name = "start:{0}, end:{1}, expected:{2}")
+  @SuppressWarnings("unused")
+  private Object[] getValidRanges() {
+    // Values are start, end, expected result
+    // For initial list of {e4, e3, e2, e1}
+    return new Object[] {
+        new Object[] {0L, 0L, new String[] {"e4"}},
+        new Object[] {0L, 1L, new String[] {"e4", "e3"}},
+        new Object[] {0L, 2L, new String[] {"e4", "e3", "e2"}},
+        new Object[] {1L, 2L, new String[] {"e3", "e2"}},
+        new Object[] {1L, -1L, new String[] {"e3", "e2", "e1"}},
+        new Object[] {1L, -2L, new String[] {"e3", "e2"}},
+        new Object[] {-2L, -1L, new String[] {"e2", "e1"}},
+        new Object[] {-1L, -1L, new String[] {"e1"}},
+        new Object[] {0L, 3L, new String[] {"e4", "e3", "e2", "e1"}},
+        new Object[] {2L, 3L, new String[] {"e2", "e1"}},
+        new Object[] {3L, 4L, new String[] {"e1"}},
+        new Object[] {0L, 4L, new String[] {"e4", "e3", "e2", "e1"}},
+        new Object[] {0L, 10L, new String[] {"e4", "e3", "e2", "e1"}},
+        new Object[] {-5L, -1L, new String[] {"e4", "e3", "e2", "e1"}},
+        new Object[] {-10L, 10L, new String[] {"e4", "e3", "e2", "e1"}}
+    };
+  }
+
+  private void initializeTestList() {
+    jedis.del(KEY);
```

Comment:
This line is not necessary, as we call `flushAll()` in the `@Before` method, which gets called before each iteration of the test, which means that this line can be removed and the method can be inlined.

Reply:
Simplified.